app/modules/listings/components/users_listings.svelte

```diff
@@ -27,10 +48,27 @@
       flash = err
     }
   }
-</script>
 
+  const fetchMore = async () => {
+    if (fetching || hasMore === false) return
+    fetching = true
+    const nextListings = await getSomeListings(offset, paginationSize)
+    if (isNonEmptyArray(nextListings)) {
+      listings = [ ...listings, ...nextListings ]
+    }
+    offset += paginationSize
+    fetching = false
+  }
+  $: {
+    if (listingBottomEl != null && hasMore) {
+      const screenBottom = windowScrollY + window.visualViewport.height
+      if (screenBottom + 100 > listingBottomEl.offsetTop) fetchMore()
```

Comment:
[non-blocking]
There is an edge case where the viewport is so big that fetching the next batch will not make the list go below the viewport bottom, meaning that `windowScrollY` will not change, and `fetchMore` will not be called anymore. This can easily be tested by setting `paginationSize = 1`